common/txmgr/types/fee_estimator.go

```diff
@@ -15,22 +16,21 @@ const (
 )
 
 // PriorAttempt provides a generic interface for reading tx data to be used in the fee esimators
-type PriorAttempt[FEE any, HASH any] interface {
+type PriorAttempt[FEE any, TX_HASH types.Hashable] interface {
```

Comment:
I don't mind doing just an interface. That keeps the overall code simple.
But it does compromise type safety. What if someone by mistake pass in a TxHash instead of a BlockHash, or an Address, etc?

I'm ok with any way which people prefer.
But I feel @jmank88 cares about type safety.

Reply:
Using an interface here _constrains_ the set of types used, limiting the set of types allowed. It is an increase in type safety over `any`. In both cases the callers still use concrete types.

Reply:
I think @samsondav meant that why can't we use types.Hashable directly as field types, and not use generics at all?

Reply:
Using a combination of interface and generics.